env0/common_template.go

```diff
@@ -0,0 +1,118 @@
+			Description: fmt.Sprintf("id of the %s", templateType),
+			Computed:    true,
+		},
+		"repository": {
+			Type:        schema.TypeString,
+			Description: fmt.Sprintf("repository url for the %s source code", templateType),
+			Required:    true,
+		},
+		"path": {
+			Type:        schema.TypeString,
+			Description: "terraform / terragrunt file folder inside source code. Should be the full path including the .yaml/.yml file",
+			Optional:    true,
+		},
+		"revision": {
+			Type:        schema.TypeString,
+			Description: "source code revision (branch / tag) to use",
+			Optional:    true,
+		},
+		"token_id": {
+			Type:        schema.TypeString,
+			Description: "the git token id to be used",
+			Optional:    true,
+		},
+		"ssh_keys": {
+			Type:        schema.TypeList,
+			Description: "an array of references to 'data_ssh_key' to use when accessing git over ssh",
+			Optional:    true,
+			Elem: &schema.Schema{
+				Type:        schema.TypeMap,
+				Description: "a map of env0_ssh_key.id and env0_ssh_key.name for each project",
+			},
+		},
+		"gitlab_project_id": {
+			Type:         schema.TypeInt,
+			Description:  "the project id of the relevant repository",
+			Optional:     true,
+			RequiredWith: []string{"token_id"},
+		},
+		"github_installation_id": {
+			Type:        schema.TypeInt,
+			Description: "the env0 application installation id on the relevant github repository",
+			Optional:    true,
+		},
+		"bitbucket_client_key": {
+			Type:        schema.TypeString,
+			Description: "the bitbucket client key used for integration",
+			Optional:    true,
+		},
+		"is_bitbucket_server": {
+			Type:        schema.TypeBool,
+			Description: fmt.Sprintf("true if this %s uses bitbucket server repository", templateType),
+			Optional:    true,
+			Default:     false,
+		},
+		"is_gitlab_enterprise": {
+			Type:        schema.TypeBool,
+			Description: fmt.Sprintf("true if this %s uses gitlab enterprise repository", templateType),
+			Optional:    true,
+			Default:     false,
+		},
+		"is_github_enterprise": {
+			Type:        schema.TypeBool,
+			Description: fmt.Sprintf("true if this %s uses github enterprise repository", templateType),
+			Optional:    true,
+			Default:     false,
+		},
+		"is_gitlab": {
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Description: fmt.Sprintf("true if this %s integrates with gitlab repository", templateType),
+			Default:     false,
+		},
+		"is_azure_devops": {
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Description: fmt.Sprintf("true if this %s integrates with azure dev ops repository", templateType),
+			Default:     false,
+		},
+	}
+
+	if templateType == CustomFlow {
+		s["name"] = &schema.Schema{
+			Type:        schema.TypeString,
+			Description: "name for the custom flow. note: for the UI to render the custom-flow please use `project-<project.id>`",
+			Required:    true,
+		}
+	}
+
+	if templateType == ApprovalPolicy {
+		s["name"] = &schema.Schema{
```

Comment:
I am allowing users to set names.
However, they must do it in the requested format.
During assignment it will fail if they do not do it in the required format and if there are any mismatches.

Reply:
I'm not sure about that part @HeverFarber can you take a look and suggest if its a good product decision in light of your latest changes?

Reply:
don't need to force the template name. let the user insert whatever name he wants

Reply:
thanks! so @TomerHeber lets remove this assertion 👍🏼